### PR TITLE
[Behat] Dehardcode the use of entities

### PR DESCRIPTION
**Pull request by GSadee**
| Q               | A
|-----------------|-----
| Branch?         | 1.14
| Bug fix?        | yes
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | 
| License         | MIT

### CODE REVIEW

**Review by mpysiak — DISMISSED** [review]

File: src/Sylius/Component/Core/Test/Services/DefaultChannelFactory.php [hunk 1]
```diff
@@ -35,15 +35,17 @@ final class DefaultChannelFactory implements DefaultChannelFactoryInterface
...
-        private RepositoryInterface $localeRepository,
-        private string $defaultLocaleCode,
+        private readonly ChannelFactoryInterface $channelFactory,
+        private readonly FactoryInterface $currencyFactory,
+        private readonly FactoryInterface $localeFactory,
+        private readonly RepositoryInterface $channelRepository,
+        private readonly RepositoryInterface $currencyRepository,
+        private readonly RepositoryInterface $localeRepository,
+        private readonly string $defaultLocaleCode,
+        private readonly FactoryInterface $shopBillingDataFactory,
```
Comment (diimpp) [inline src/Sylius/Component/Core/Test/Services/DefaultChannelFactory.php]: Just a suggestion
```suggestion
        private readonly FactoryInterface $shopBillingDataFactory,
        private readonly string $defaultLocaleCode,
```

File: src/Sylius/Component/Core/spec/Test/Services/DefaultChannelFactorySpec.php [hunk 2]
```diff
@@ -56,9 +58,11 @@ function it_creates_a_default_channel_and_persist_it(
         RepositoryInterface $channelRepository,
         RepositoryInterface $currencyRepository,
         RepositoryInterface $localeRepository,
+        FactoryInterface $shopBillingDataFactory,
```
Comment (diimpp) [inline src/Sylius/Component/Core/spec/Test/Services/DefaultChannelFactorySpec.php]: Please put it at line 58 with other factories.

image

**Review by diimpp — DISMISSED** [review]

File: src/Sylius/Component/Core/Test/Services/DefaultChannelFactory.php (hunk 1 above)
Comment (GSadee) [reply src/Sylius/Component/Core/Test/Services/DefaultChannelFactory.php]: Although I agree with you and I would also prefer a different order, I don't want to change the order of the services in the constructor here.

File: src/Sylius/Component/Core/spec/Test/Services/DefaultChannelFactorySpec.php (hunk 2 above)
Comment (GSadee) [reply src/Sylius/Component/Core/spec/Test/Services/DefaultChannelFactorySpec.php]: I want to keep here the same order as it is in the constructor

**Review by Wojdylak — APPROVED** [review]

**Comment by Wojdylak** [comment]
Thank you, @GSadee!

**Pull request merged**